src/Domain/Transaction/Factory/TransactionFactory.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PhpFinance\DoubleEntry\Domain\Transaction\Factory;
+
+use DateTimeImmutable;
+use PhpFinance\DoubleEntry\Domain\Posting\Factory\PostingFactory;
+use PhpFinance\DoubleEntry\Domain\Transaction\Transaction;
+
+final class TransactionFactory
+{
+    public function __construct(
+        private TransactionIdFactory $transactionIdFactory,
+        private PostingFactory $postingFactory,
+    ) {
+    }
+
+    public function create(
+        DateTimeImmutable $date,
+        PostingData $postingData,
+        PostingData ...$postingsData,
```

Comment:
Looks interesting. What's intent here?

Reply:
This ensures that at least one posting will be passed.